src/components/ColorCard.svelte

```diff
@@ -33,22 +38,38 @@
     colorName = nearest(color.toString('hex')).name
   }
 
-  const onColorInput = async (event) => {
-    colorInput = event.target.value
-    color = new Color(colorInput)
-    inputFormat = new TinyColor(colorInput).format
+  /*
+    Check if passed argument is a valid color and update `isValidColor` accordingly
+  */
+  const validateColor = (value: string) => isValidColor = chroma.valid(value)
+
+  /*
+    Set passed color as active if it's a valid color
+  */
+  const setNewColor = async (newColor: string) => {
+    validateColor(newColor)
+    if (!isValidColor) return
+
+    color = new Color(newColor)
+    inputFormat = new TinyColor(newColor).format
     findColorName()
 
     dispatch('change', {
       color,
     })
   }
 
+  /*
+    Proceed with color change if the user's input isn't empty, otherwise - set default color as active
+  */
+  const onColorInputChange = () => colorInput.value ? setNewColor(colorInput.value) : setNewColor(defaultColor)
+  
   const findColorNameThrottled = throttle(() => {
     colorName = nearest(color.toString('hex')).name
   }, 400)
 
   const onColorPickerUpdate = () => {
+    validateColor(colorInput.value) // Udpate `isValidColor`
```

Comment:
The color picker will always produce a valid color so we don't need to perform this check here. Simply setting the `isValidColor` to `true` if it isn't already would be enough here.

Reply:
On opening of color picker the update event is fired and even without changing any value this will set `isValidColor` to true

Reply:
I went with other solution to this problem and removed this line during fixing other bug.